administrator/components/com_menus/Model/ItemsModel.php

```diff
@@ -495,6 +496,12 @@ protected function getListQuery()
 			$query->where('a.language = ' . $db->quote($language));
 		}
 
+		// Filter on componentName
+		if ($componentName = $this->getState('filter.componentName'))
+		{
+			$query->where('e.name = ' . $db->quote($componentName));
```

Comment:
should be full quoted
```suggestion
			$query->where($db->quoteName('e.name') . ' = ' . $db->quote($componentName));
```